docs/reference_return_entity.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: api_reference
+title: API Reference
+permalink: /spec/return_entity
+index: 2
+---
+
+# Return Entity
+
+This page describes returning the entity for resource methods that are not originally intended to return the entity.
+For example, returning the entity is normal behavior for GET and FINDER, so this page does not apply to them.
+
+For methods such as CREATE, however, the created entity is not returned in the response because the client
+already has the entity when sending the request. Despite this, there are use cases where the server will
+attach additional data to the new entity. Returning the entity in the CREATE response saves the client
+from having to make an extra GET request.
+
+## Contents
+
+- 	[Supported Methods](#supported-methods)
+-   [How to Enable](#how-to-enable)
+-	[Query Parameter](#query-parameter)
+
+## Supported Methods
+
+Currently, this extra functionality is supported for the following resource methods:
+
+- CREATE
+- PARTIAL_UPDATE
+- BATCH_CREATE
+
+## How to Enable
+
+To enable returning the entity for a given resource method, there are two requirements
+that must be fulfilled.
+
+First, the resource method must be annotated with the @`ReturnEntity` annotation.
+This applies to all resource methods.
+
+Second, the return type of the method must be a valid "Return Entity" return type.
+This is specific to each resource method. The following table lists which "Return Entity"
+return type corresponds to which resource method:
+
+| Resource Method | Standard Return Type | "Return Entity" Return Type |
+|-----------------|----------------------|-----------------------------|
+| CREATE          | `CreateResponse`     | `CreateKVResponse`          |
+| PARTIAL_UPDATE  | `UpdateResponse`     | `UpdateEntityResponse`      |
+| BATCH_CREATE    | `BatchCreateResult`  | `BatchCreateKVResult`       |
+
+Here is an example method signature for a CREATE resource method that will enable the entity to be returned.
+Note how both the annotation and the required return type are present:
+
+```java
+@ReturnEntity
+public CreateKVResponse create(V entity);
+```
+
+If both of these requirements are fulfilled, then the entity will be returned in the response by default.
+
+## Query Parameter
```

Comment:
I think "Client specified behavior" or "Changing default behavior" might be a better heading than query parameter.

Reply:
I considered this too. Especially if I add info about `shouldReturnEntity`, then "query parameter" would be too specific. Thanks